test/test_rocfft.cpp

```diff
@@ -0,0 +1,380 @@
+#define BOOST_TEST_MODULE TestrocFFT
+#include <iostream>
+#include <vector>
+#include <complex>
+
+#include <cstdint>
+#include "hip/hip_runtime_api.h"
+#include "hip/hip_vector_types.h"
+#include "rocfft.h"
+#include <boost/test/unit_test.hpp>
+#include <iostream>
+
+BOOST_AUTO_TEST_CASE( FFT1DSmall, * boost::unit_test::tolerance(0.0001f) )
+{
+  // rocFFT gpu compute
+        // ========================================
+
+        rocfft_setup();
+
+        std::size_t N = 2 << 10; //should be small enough to not need a work_buffer
+        std::size_t Nbytes = N * sizeof(float2);
+
+        // Create HIP device buffer
+        float2 *x;
+        hipMalloc(&x, Nbytes);
```

Comment:
can  you add error-checkers for the HIP/rocfft commands?

Reply:
fixed f10eea4